src/thor/bidirectional_astar.cc

```diff
@@ -181,6 +181,10 @@ void BidirectionalAStar::ExpandForward(GraphReader& graphreader,
                                      (pred.not_thru_pruning() || !directededge->not_thru()));
     adjacencylist_forward_->add(idx);
     *es = {EdgeSet::kTemporary, idx};
+
+    // setting this edge as reached
+    if (expansion_callback_)
```

Comment:
`std::function<>` has an `operator bool` which lets you check if its actually assigned to something callable, this check is way faster than `std::weak_ptr<>::lock`